frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/stateUpdate/vmSettings/vm-settings-tab-state-update.ts

```diff
@@ -126,6 +128,23 @@ export const osUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions)
   }
 };
 
+export const nativeK8sUpdater = ({ id, dispatch, getState, changedCommonData }: UpdateOptions) => {
+  const state = getState();
+  if (!changedCommonData.has(VMWizardProps.openshiftFlag)) {
+    return;
+  }
+  const openshiftFlag = state[featureReducerName].get(FLAGS.OPENSHIFT);
+
+  dispatch(
+    vmWizardInternalActions[InternalActionType.UpdateVmSettings](id, {
+      [VMSettingsField.WORKLOAD_PROFILE]: {
+        isHidden: { VM_SETTINGS_METADATA_ID: !openshiftFlag },
```

Comment:
isHidden should be annotated by something else than VM_SETTINGS_METADATA_ID to prevent collisions, I guess you could use `FLAGS.OPENSHIFT`